pageserver/src/tenant/timeline.rs

```diff
@@ -1056,72 +1059,78 @@ impl Timeline {
         //    b) typical case now is that there is nothing to sync, this downloads a lot
         //       1) if there was another pageserver that came and generated new files
         //       2) during attach of a timeline with big history which we currently do not do
-        for path in remote_filenames.difference(&local_filenames) {
-            let fname = path.to_str().unwrap();
-            info!("remote layer file {fname} does not exist locally");
+        for fname in remote_filenames.difference(&local_filenames) {
+            info!(
+                "remote layer file {} does not exist locally",
```

Comment:
Good that your PR makes such things more obvious: `download_missing` function is rather generic and does not print tenant/timeline ids, layer names are also rather uninformative, so just printing the layer name, we would not know in which timeline it is missing.

If possible, we might want to print the full path here.

Reply:
> and does not print tenant/timeline ids

And it doesnt need to. Because they are included into parent span variables.

Reply:
Found from the 2nd try, the span is also split in two, as I see.
Not sure I want to check all other such places, so will leave the other comment unresolved.

Reply:
I agree that it is a bit detached with spans, but to me thats better than carrying ids and formatting them for every message